crates/consensus/src/hardforks/traits.rs

```diff
@@ -0,0 +1,9 @@
+//! The trait abstraction for a Hardfork.
+
+use alloy_primitives::Bytes;
+
+/// The trait abstraction for a Hardfork.
+pub trait Hardfork {
+    /// Returns the hardfork upgrade transactions as [Bytes].
+    fn txs(&self) -> impl Iterator<Item = Bytes> + '_;
+}
```

Comment:
how about returning the `TxDeposit` instead and letting the caller encode it

Reply:
the caller is likely only interested in the raw txs, but it could be useful to have an additional fn that returns tx envelopes

Reply:
would be nice with an alloy repo for rollup abstractions. this trait is likely relevant for other evm rollups, but not for l1 alloy.

Reply:
Pushed the changes - let me know if this is what you're looking for :)